### ISSUE TITLE
Automated accessibility testing

### ISSUE THREAD

**Issue by dylans** [theintern/intern#384]
The two more popular approaches to automate accessibility testing currently seem to be relying on Chrome Accessibility Developer Tools or a service such as tenon.io . Google's tools provide a limited set of automated options compared to tenon.

We should consider what we want to add in the future to provide some level of automated accessibility testing.

**Comment by csnover** [comment]
What does this mean?

**Comment by kitsonk** [comment]
I suspect this means providing an out of the box "wrapper" to the [audit API for Google Chrome's Accessibility Developer Tools](https://github.com/GoogleChrome/accessibility-developer-tools#using-the-audit-api) to "audit" any application and pass or fail on certain thresholds.

**Comment by dylans** [comment]
Yes, what @kitsonk said for DevTools a11y testing. For tenon.io, they have an API ( e.g. http://tenon.io/documentation/understanding-request-parameters.php ) to run accessibility tests and their API is the best option I've seen that's out there currently. If we're interested in this, we can reach out to them and see what they offer for open source projects that want to integrate with their API.

The thought process behind all of this is that a11y testing takes time, and if there's an opportunity to automate the verification of an application being accessible, we should try to offer that as part of a future Intern release.

**Comment by sholladay** [comment]
There's a Node.js API for this made by sindresorhus and addyosmani that could be automated pretty easily. https://github.com/addyosmani/a11y

**Comment by ajaswa** [comment]
I've put together an example of how I got automated accessibility testing working **inside** a test suite. It's **not** pretty and I'm working on abstracting it out into a helper and to use other things in Intern but it seems to work for now. It uses Deque Labs aXe core and Google Chrome Accessibility Developer Tools.

https://github.com/ajaswa/a11y-intern-example

I hope it helps someone.

**Comment by jason0x43** [comment]
This functionality was added via https://github.com/theintern/intern-a11y
